apollo-router/src/plugins/telemetry/otlp/mod.rs

```diff
@@ -47,6 +58,67 @@ impl Exporter {
             Exporter::Http(exporter) => Ok(exporter.clone().unwrap_or_default().exporter()?.into()),
         }
     }
+
+    pub fn metrics_exporter(&self) -> Result<PushController, ConfigurationError> {
+        match &self {
+            #[cfg(feature = "otlp-grpc")]
+            Exporter::Grpc(exporter) => {
+                let mut export_config = opentelemetry_otlp::ExportConfig::default();
+                if let Some(grpc_exporter_cfg) = exporter {
+                    if let Some(endpoint) = &grpc_exporter_cfg.export_config.endpoint {
+                        export_config.endpoint = endpoint.clone().to_string();
+                    }
+                    if let Some(timeout) = &grpc_exporter_cfg.export_config.timeout {
+                        export_config.timeout = Duration::from_secs(*timeout);
+                    }
+                    if let Some(protocol) = &grpc_exporter_cfg.export_config.protocol {
+                        export_config.protocol = *protocol;
+                    }
+                }
+
+                let push_ctrl = opentelemetry_otlp::new_pipeline()
+                    .metrics(tokio::spawn, delayed_interval)
+                    .with_exporter(
+                        opentelemetry_otlp::new_exporter()
+                            .tonic()
+                            .with_export_config(export_config),
+                    )
+                    .with_aggregator_selector(selectors::simple::Selector::Exact)
+                    .build()?;
+
+                Ok(push_ctrl)
+            }
+            #[cfg(feature = "otlp-http")]
+            Exporter::Http(_exporter) => {
+                // let mut export_config = opentelemetry_otlp::ExportConfig::default();
+                // if let Some(http_exporter_cfg) = exporter {
+                //     if let Some(endpoint) = &http_exporter_cfg.export_config.endpoint {
+                //         export_config.endpoint = endpoint.clone().to_string();
+                //     }
+                //     if let Some(timeout) = &http_exporter_cfg.export_config.timeout {
+                //         export_config.timeout = Duration::from_secs(*timeout);
+                //     }
+                //     if let Some(protocol) = &http_exporter_cfg.export_config.protocol {
+                //         export_config.protocol = *protocol;
+                //     }
+                // }
+
+                // let push_ctrl = opentelemetry_otlp::new_pipeline()
+                //     .metrics(tokio::spawn, delayed_interval)
+                //     .with_exporter(
+                //         opentelemetry_otlp::new_exporter()
+                //             .http()
+                //             .with_export_config(export_config),
+                //     )
+                //     .with_aggregator_selector(selectors::simple::Selector::Exact)
+                //     .build()?;
+
+                // Ok(push_ctrl)
```

Comment:
do we want to keep this comment around? why ?

Reply:
I think it could be interesting to keep it. When the issue will be resolved we would just need to uncomment and it should work

Reply:
sgtm